packages/flutter_tools/lib/src/base/context.dart

```diff
@@ -3,78 +3,39 @@
 // found in the LICENSE file.
 
 import 'dart:async';
+import 'package:stack_trace/stack_trace.dart';
 
 final AppContext _defaultContext = new AppContext();
 
 typedef void ErrorHandler(dynamic error, StackTrace stackTrace);
 
-/// A singleton for application functionality. This singleton can be different
-/// on a per-Zone basis.
-AppContext get context {
-  AppContext currentContext = Zone.current['context'];
-  return currentContext == null ? _defaultContext : currentContext;
-}
+AppContext _context = _defaultContext;
+
+/// A singleton for the application functionality.
+AppContext get context => _context;
 
 class AppContext {
   Map<Type, dynamic> _instances = <Type, dynamic>{};
-  Zone _zone;
-
-  bool isSet(Type type) {
-    if (_instances.containsKey(type))
-      return true;
 
-    AppContext parent = _calcParent(Zone.current);
-    return parent != null ? parent.isSet(type) : false;
-  }
-
-  dynamic getVariable(Type type) {
-    if (_instances.containsKey(type))
-      return _instances[type];
+  bool isSet(Type type) => _instances.containsKey(type);
 
-    AppContext parent = _calcParent(_zone ?? Zone.current);
-    return parent?.getVariable(type);
-  }
+  dynamic getVariable(Type type) => _instances[type];
 
-  void setVariable(Type type, dynamic instance) {
-    _instances[type] = instance;
-  }
+  void setVariable(Type type, dynamic instance) => _instances[type] = instance;
```

Comment:
The style guide [advises against](https://www.dartlang.org/guides/language/effective-dart/usage#consider-using--for-short-members-whose-body-is-a-single-return-statement) using `=>` for void functions.